route/v1/types.go

```diff
@@ -145,8 +146,131 @@ type RouteSpec struct {
 	// +kubebuilder:validation:Enum=None;Subdomain;""
 	// +kubebuilder:default=None
 	WildcardPolicy WildcardPolicyType `json:"wildcardPolicy,omitempty" protobuf:"bytes,7,opt,name=wildcardPolicy"`
+
+	// httpHeaders defines policy for HTTP headers.
+	//
+	// If this field is empty, the default values are used.
+	//
+	// +optional
+	HTTPHeaders RouteHTTPHeaders `json:"httpHeaders,omitempty" protobuf:"bytes,9,opt,name=httpHeaders"`
+}
+
+// RouteHTTPHeaders defines policy for HTTP headers.
+type RouteHTTPHeaders struct {
+	// actions specifies actions to take on headers.
+	// Note that this option only applies to cleartext HTTP connections
+	// and to secure HTTP connections for which the ingress controller
+	// terminates encryption (that is, edge-terminated or reencrypt
+	// connections).  Headers cannot be set or deleted for TLS passthrough
+	// connections.
+	// Setting HSTS header is supported via another mechanism in Ingress.Spec.RequiredHSTSPolicies.
+	// The header set using route will override the header set in ingress controller if the headers defined both were the same.
```

Comment:
nit - I think this is clearer, if you end up making any other changes please change this.
```suggestion
	// A header set/deleted using route will override the header of the same name set/deleted in ingress controller.
```

Reply:
Oops, in past reviews, I've been trying to get Miheer not to use slashes.  Do you prefer we use slashes?

Reply:
I think for godoc I would spell it out rather than using the slashes, but I'm not that strongly opinionated on this one, or, you can probably avoid altogether
```suggestion
	// Any header configuration applied directly via a Route resource will override header configuration
	// for a header of the same name applied via an IngressController resource.
```

Reply:
As long as it doesn't just say "set".  I think @JoelSpeed's suggestion is best.

Reply:
Added Joels's suggestion with addition route annotation.

Reply:
Did you push the update?  I still see the fold text: https://github.com/openshift/api/blob/8e4d36aaebb33d8bc4e10bd2e5cfc2efeb803ff3/route/v1/types.go#L167

Reply:
https://github.com/openshift/api/compare/8e4d36aaebb33d8bc4e10bd2e5cfc2efeb803ff3..3ed233511a8dd9484a47a184709f6a75d32ce70e implemented Joel's suggestion.  Thanks!